pkg/arm/v6/data/master-startup.sh

```diff
@@ -1,5 +1,54 @@
 #!/bin/bash -ex
 
+{{ if gt (len .Config.SecurityPatchPackages) 0 }}
+logger -t master-startup.sh "installing red hat cdn configuration on $(hostname)"
+cat >/var/lib/yum/client-cert.pem <<'EOF'
+{{ CertAsBytes .Config.Certificates.PackageRepository.Cert | String }}
+EOF
+cat >/var/lib/yum/client-key.pem <<'EOF'
+{{ PrivateKeyAsBytes .Config.Certificates.PackageRepository.Key | String }}
+EOF
+# TODO: delete the following section after all clusters are run with a vm image with kickstart.repo baked in
+cat > /etc/yum.repos.d/kickstart.repo <<'EOF'
+[rhel-7-server-rpms]
+name=Red Hat Enterprise Linux 7 Server (RPMs)
+baseurl=https://cdn.redhat.com/content/dist/rhel/server/7/7Server/$basearch/os
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+sslcacert=/etc/rhsm/ca/redhat-uep.pem
+sslclientcert=/var/lib/yum/client-cert.pem
+sslclientkey=/var/lib/yum/client-key.pem
+enabled=yes
+
+[rhel-7-server-extras-rpms]
+name=Red Hat Enterprise Linux 7 Server - Extras (RPMs)
+baseurl=https://cdn.redhat.com/content/dist/rhel/server/7/7Server/$basearch/extras/os
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+sslcacert=/etc/rhsm/ca/redhat-uep.pem
+sslclientcert=/var/lib/yum/client-cert.pem
+sslclientkey=/var/lib/yum/client-key.pem
+enabled=yes
+
+[rhel-7-server-ose-3.11-rpms]
+name=Red Hat OpenShift Container Platform 3.11 (RPMs)
+baseurl=https://cdn.redhat.com/content/dist/rhel/server/7/7Server/$basearch/ose/3.11/os
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+sslcacert=/etc/rhsm/ca/redhat-uep.pem
+sslclientcert=/var/lib/yum/client-cert.pem
+sslclientkey=/var/lib/yum/client-key.pem
+enabled=yes
+EOF
+yum makecache
+
+{{ range .Config.SecurityPatchPackages }}
+logger -t master-startup.sh "installing {{.}} on $(hostname)"
+yum install -y {{.}}
+{{end}}
```

Comment:
need to see the code that will handle reboots.  Also close on error handling/reporting.

Reply:
added code to handle reboots and retries to the `yum install` command. Note that the reboot is handled in a trap. This is because if the reboot is done immediately when we determine it is needed then the rest of the waagent script will not be run and the cluster wont come up. Another option is to move the security updates code to the bottom of the script however I think updates should be applied to the system before bring up the control plane.